src/ipred.rs

```diff
@@ -364,6 +364,29 @@ impl DcGen {
     }
 }
 
+/// Reconstructs the reference to the topleft edge array from a pointer into the
+/// array and an offset from the start of the array.
+///
+/// The topleft pointer passed to asm is always a pointer into a buffer of
+/// length [`SCRATCH_EDGE_LEN`]. For the Rust fallbacks we also pass in the
+/// offset from the front of the buffer so that we can reconstruct the original
+/// array reference in order to use safe array operations within the fallbacks.
+///
+/// # Safety
+///
+/// `topleft_ptr` must be a pointer into an array of length [`SCRATCH_EDGE_LEN`]
+/// and is `topleft_off` elements from the beginning of the array. This should
+/// be guaranteed by the logic in `angular_ipred::call`.
```

Comment:
```suggestion
/// be guaranteed by the logic in [`angular_ipred::Fn::call`].
```

Reply:
Ugh, I accidently merged this before pushing the change that fixed this, sorry. Can we live with this not being a link in the doc comment? I can put up a separate PR for that, but tbh that doesn't seem worth the effort to me.

Reply:
Yeah, sure, or you can just tack it onto some other commit.